scylla-cql/src/value.rs

```diff
@@ -779,3 +791,461 @@ pub struct CqlDuration {
+            _ => None,
+        }
+    }
+
+    pub fn as_uuid(&self) -> Option<Uuid> {
+        match self {
+            Self::Uuid(u) => Some(*u),
+            _ => None,
+        }
+    }
+
+    pub fn as_float(&self) -> Option<f32> {
+        match self {
+            Self::Float(f) => Some(*f),
+            _ => None,
+        }
+    }
+
+    pub fn as_int(&self) -> Option<i32> {
+        match self {
+            Self::Int(i) => Some(*i),
+            _ => None,
+        }
+    }
+
+    pub fn as_bigint(&self) -> Option<i64> {
+        match self {
+            Self::BigInt(i) => Some(*i),
+            _ => None,
+        }
+    }
+
+    pub fn as_tinyint(&self) -> Option<i8> {
+        match self {
+            Self::TinyInt(i) => Some(*i),
+            _ => None,
+        }
+    }
+
+    pub fn as_smallint(&self) -> Option<i16> {
+        match self {
+            Self::SmallInt(i) => Some(*i),
+            _ => None,
+        }
+    }
+
+    pub fn as_blob(&self) -> Option<&Vec<u8>> {
+        match self {
+            Self::Blob(v) => Some(v),
+            _ => None,
+        }
+    }
+
+    pub fn as_text(&self) -> Option<&String> {
+        match self {
+            Self::Text(s) => Some(s),
+            _ => None,
+        }
+    }
+
+    pub fn as_timeuuid(&self) -> Option<CqlTimeuuid> {
+        match self {
+            Self::Timeuuid(u) => Some(*u),
+            _ => None,
+        }
+    }
+
+    pub fn into_string(self) -> Option<String> {
+        match self {
+            Self::Ascii(s) => Some(s),
+            Self::Text(s) => Some(s),
+            _ => None,
+        }
+    }
+
+    pub fn into_blob(self) -> Option<Vec<u8>> {
+        match self {
+            Self::Blob(b) => Some(b),
+            _ => None,
+        }
+    }
+
+    pub fn as_inet(&self) -> Option<IpAddr> {
+        match self {
+            Self::Inet(a) => Some(*a),
+            _ => None,
+        }
+    }
+
+    pub fn as_list(&self) -> Option<&Vec<CqlValue>> {
+        match self {
+            Self::List(s) => Some(s),
+            _ => None,
+        }
+    }
+
+    pub fn as_set(&self) -> Option<&Vec<CqlValue>> {
+        match self {
+            Self::Set(s) => Some(s),
+            _ => None,
+        }
+    }
+
+    pub fn as_map(&self) -> Option<&Vec<(CqlValue, CqlValue)>> {
+        match self {
+            Self::Map(s) => Some(s),
+            _ => None,
+        }
+    }
+
+    pub fn as_udt(&self) -> Option<&Vec<(String, Option<CqlValue>)>> {
+        match self {
+            Self::UserDefinedType { fields, .. } => Some(fields),
+            _ => None,
+        }
+    }
+
+    pub fn into_vec(self) -> Option<Vec<CqlValue>> {
+        match self {
+            Self::List(s) => Some(s),
+            Self::Set(s) => Some(s),
+            _ => None,
+        }
+    }
+
+    pub fn into_pair_vec(self) -> Option<Vec<(CqlValue, CqlValue)>> {
+        match self {
+            Self::Map(s) => Some(s),
+            _ => None,
+        }
+    }
```

Comment:
:thought_balloon: :camping: Wouldn't we prefer to return `Result<CastT, Self>`, not to lose `self` in case of conversion failure?

Reply:
I think this is the only comment that we can't really address after 1.0 (which doesn't mean we have to address it now - we could to that for 2.0). Would you like to open an issue?